modules/apps/redirect/redirect-web/src/main/resources/META-INF/resources/js/RedirectManagementToolbarPropsTransformer.js

```diff
@@ -12,28 +12,33 @@
  * details.
  */
 
-import {DefaultEventHandler} from 'frontend-js-web';
-import {Config} from 'metal-state';
+import {postForm} from 'frontend-js-web';
 
-class RedirectManagementToolbarDefaultEventHandler extends DefaultEventHandler {
-	deleteSelectedRedirectEntries() {
-		const form = this.one('#fm');
+export default function propsTransformer({
+	additionalProps: {deleteRedirectEntriesURL},
+	portletNamespace,
+	...otherProps
+}) {
+	const deleteSelectedRedirectEntries = () => {
+		const form = document.getElementById(`${portletNamespace}fm`);
 
-		Liferay.Util.postForm(form, {
+		postForm(form, {
 			data: {
 				deleteEntryIds: Liferay.Util.listCheckedExcept(
 					form,
-					this.ns('allRowIds')
+					`${portletNamespace}allRowIds`
 				),
 			},
-			url: this.deleteRedirectEntriesURL,
+			url: deleteRedirectEntriesURL,
 		});
-	}
-}
-
-RedirectManagementToolbarDefaultEventHandler.STATE = {
-	deleteRedirectEntriesURL: Config.string(),
-	spritemap: Config.string(),
-};
+	};
 
-export default RedirectManagementToolbarDefaultEventHandler;
+	return {
+		...otherProps,
+		onActionButtonClick: (event, {item}) => {
+			if (item.data.action === 'deleteSelectedRedirectEntries') {
```

Comment:
Does item always have a `data` property? If not this will fail in that case

Reply:
Totally agree according to this is optional https://github.com/liferay-lima/liferay-portal/pull/1445#discussion_r569315331

Reply:
We already have discussed it and as Julien mentioned here https://github.com/liferay-frontend/liferay-portal/pull/757#discussion_r570200021, data is always set ini the java [code](https://github.com/liferay-frontend/liferay-portal/blob/5143e1ca77ded386e0edb3d1f9c1069c09b51290/modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/display/context/RedirectManagementToolbarDisplayContext.java#L66-L75)

Reply:
So the discussion boils down to this: Is there a legitimate use when `item.data` would not exist? Could there be an action dropdown item that doesn't represent an action? Is there another mechanism to wire action apart from `item.data.action`? 

If the only reason to have null-check is to prevent developer mistakenly forgetting to set `dropdownItem.putData("action", ...)`, then I'd say it is actually better to fail with error than to recover gracefully. Existing code in v2 has the check, but I'd consider code in this PR to be an improvement.

Reply:
Makes sense to me, thanks @ambrinchaudhary @markocikos for the explanation.

@carloslancha if you are ok, resolve and forward it :D

Reply:
What if the item is a link?

Reply:
> What if the item is a link? Is that possible?

@carloslancha Answering this led me to a bit of a rabbit hole :slightly_smiling_face: 

I already had an answer ready: If the item is a link, then we [don't have `onClick` ](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ActionControls.js#L40-L62)and we never reach code in `PropsTransformer`. All nicely wrapped up.

... however, this is only true for quick actions under tag controls, not for items in the dropdown. They are under Clay control, and in Clay we can have [a link with onClick](https://github.com/liferay/clay/blob/master/packages/clay-drop-down/src/Item.tsx#L67-L79).

Thinking back, I don't remember why I chose not to put click event on a link. At some point, I accepted as a fact, although:
- [Toolbar v2 implementation](https://github.com/liferay/clay/blob/2.x/packages/clay-dropdown/src/ClayDropdownBase.soy#L266) has it 
- Clay component has it.
- A case like this can easily exist, where code modifies `href` before navigating to it.

With all this in mind, what we should do it is:
1. Add [this block](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ActionControls.js#L54-L58) to `<ClayLink>` above.
2. Keep optional chaining in `item.data?.action`

@ambrinchaudhary please add the change to this PR.

Apologies to everybody involved for the confusion :disappointed: 

/cc @julien @jonmak08 @boton

Reply:
Nice analysis @markocikos ! Thx!!

Reply:
Thanks for sharing the conclusion @markocikos